src/components/app/data/services/programs.test.js

```diff
@@ -91,17 +101,114 @@ describe('fetchLearnerProgramProgressDetail', () => {
   });
 
   it('returns the api call with a 200', async () => {
-    axiosMock.onGet(PROGRAM_PROGRESS_DETAIL_URL).reply(200, mockResponse);
+    axiosMock.onGet(PROGRAM_PROGRESS_DETAIL_URL).reply(200, mockProgramProgressResponse);
 
     const result = await fetchLearnerProgramProgressDetail(mockProgramUUID);
-    expect(result).toEqual(mockResponse);
+    expect(result).toEqual(mockProgramProgressResponse);
   });
   it('returns the api call with a 404 and logs an error', async () => {
-    axiosMock.onGet(PROGRAM_PROGRESS_DETAIL_URL).reply(404, mockResponse);
+    axiosMock.onGet(PROGRAM_PROGRESS_DETAIL_URL).reply(404, mockProgramProgressResponse);
 
     const result = await fetchLearnerProgramProgressDetail(mockProgramUUID);
     expect(logError).toHaveBeenCalledTimes(1);
     expect(logError).toHaveBeenCalledWith(new Error('Request failed with status code 404'));
-    expect(result).toEqual(mockEmptyResponse);
+    expect(result).toEqual(mockEmptyProgressResponse);
+  });
+});
+
+describe('fetchProgramDetails', () => {
+  const ENTERPRISE_UUID = '12345678-9000-0000-0000-123456789101';
+  const PROGRAM_UUID = '12345678-9000-1111-1111-123456789101';
+  const PROGRAM_API_ENDPOINT = `${APP_CONFIG.DISCOVERY_API_BASE_URL}/api/v1/programs/${PROGRAM_UUID}/`;
+  const COURSE_KEY = 'edX+DemoX';
+  const COURSE_KEY2 = 'edX+WowX';
+  const PROGRAM_DATA = {
+    courses: [
+      {
+        key: COURSE_KEY,
+        uuid: '12345678-be4c-4e1d-0000-2d60323db911',
+        title: 'Introduction to Cloud Computing',
+        activeCourseRun: undefined,
+        course_runs: [
+          {
+            key: 'course-v1:edX+DemoX+2T2020',
+            uuid: '12345678-be4c-4e1d-0000-2d60323db911',
+            title: 'Introduction to Cloud Computing',
+            short_description: 'course run description',
+            start: '2020-07-21T16:00:00Z',
+          },
+        ],
+        short_description: 'course description',
+      },
+      {
+        key: COURSE_KEY2,
+        uuid: '12345678-be4c-4e1d-0000-2d60323db911',
+        title: 'Introduction to Cloud Computing',
+        activeCourseRun: undefined,
+        course_runs: [
+          {
+            key: 'course-v1:edX+DemoX+2T2020',
+            uuid: '12345678-be4c-4e1d-0000-2d60323db911',
+            title: 'Introduction to Cloud Computing',
+            short_description: 'course run description',
+            start: '2020-07-21T16:00:00Z',
+          },
+        ],
+        short_description: 'course description',
+      },
+    ],
+    catalogContainsProgram: true,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    axiosMock.resetHistory();
+  });
+
+  global.structuredClone = val => JSON.parse(JSON.stringify(val));
```

Comment:
nit: I wonder if it might make sense to do this mock of `structureClone` within `setupTest.js` so it become applicable to all Jest tests throughout the app (e.g., could see `structuredClone` being used in other places in the future, too).